File: frontend/src/model/Delivery.ts
export default class Delivery {
    readonly id: string | null;
    readonly targetCustomerID: string;
    readonly targetBoxID: string;
    readonly delivererID: string;
    readonly active: boolean;
    readonly status: string;

    constructor(id: string | null, targetCustomerID: string, targetBoxID: string, delivererID: string, status: string, active: boolean) {
        this.id = id;
        this.targetCustomerID = targetCustomerID;
        this.targetBoxID = targetBoxID;
        this.delivererID = delivererID;
        this.status = status;
        this.active = active;
    }

    static fromJson(json: any): Delivery {
        return new Delivery(json.id, json.targetCustomerID, json.targetBoxID, json.delivererID, json.status, json.active);
    }
}

File: microservices/delivery-service/src/main/java/com/group40/deliveryservice/model/Delivery.java
package com.group40.deliveryservice.model;

import org.springframework.data.annotation.Id;

public class Delivery {
    @Id
    private String id;

    private boolean active;
    private String targetCustomerID;
    private String targetBoxID;
    private String delivererID;
    private DeliveryStatus status;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public boolean isActive() {
        return active;
    }

    public void setActive(boolean active) {
        this.active = active;
    }

    public String getTargetCustomerID() {
        return targetCustomerID;
    }

    public void setTargetCustomerID(String targetCustomerID) {
        this.targetCustomerID = targetCustomerID;
    }

    public String getTargetBoxID() {
        return targetBoxID;
    }

    public void setTargetBoxID(String targetBoxID) {
        this.targetBoxID = targetBoxID;
    }

    public String getDelivererID() {
        return delivererID;
    }

    public void setDelivererID(String delivererID) {
        this.delivererID = delivererID;
    }

    public DeliveryStatus getStatus() {
        return status;
    }

    public void setStatus(DeliveryStatus status) {
        this.status = status;
    }
}
